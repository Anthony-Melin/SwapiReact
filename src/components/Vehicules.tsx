import { PropsWithChildren } from 'react'

type VehiculesProps = PropsWithChildren<{ values: string[] }>

const Vehicules = ({ values, ...restProps }: VehiculesProps) => {
    return (
        <div datatest-id="vehicules" {...restProps}>
            <p>Vehicules</p>
            <ul>
                {values.map((value, index) => {
                    return <li datatest-id={`vehicule-${index}`}>{value}</li>
                })}
            </ul>
        </div>
    )
}

export default Vehicules
