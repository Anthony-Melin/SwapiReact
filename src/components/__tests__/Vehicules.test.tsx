import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Vehicules from '../Vehicules.tsx'

describe('Vehicules', () => {
    describe('with empty list', () => {
        beforeEach(() => {
            render(<Vehicules values={[]} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('vehicules')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('vehicules')).toHaveTextContent('Vehicules')
        })

        it('should have empty list label', () => {
            expect(screen.getByTestId('vehicules-empty')).toHaveTextContent('No vehicules linked')
        })
    })

    describe('with filled list', () => {
        beforeEach(() => {
            render(<Vehicules values={['VALUE_1', 'VALUE_2']} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('vehicules')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('vehicules')).toHaveTextContent('Vehicules')
        })

        it('should contains elements', () => {
            expect(screen.getByTestId('vehicules-0')).toHaveTextContent('VALUE_1')
            expect(screen.getByTestId('vehicules-1')).toHaveTextContent('VALUE_2')
        })
    })
})
