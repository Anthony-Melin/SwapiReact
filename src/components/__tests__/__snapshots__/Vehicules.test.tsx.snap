// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Vehicules > with empty list > should match snapshot 1`] = `
<div
  data-testid="vehicules"
>
  <p
    data-testid="vehicules-title"
  >
    Vehicules
  </p>
  <p
    data-testid="vehicules-empty"
  >
    No 
    vehicules
     linked
  </p>
</div>
`;

exports[`Vehicules > with filled list > should match snapshot 1`] = `
<div
  data-testid="vehicules"
>
  <p
    data-testid="vehicules-title"
  >
    Vehicules
  </p>
  <ul>
    <li
      data-testid="vehicules-0"
    >
      VALUE_1
    </li>
    <li
      data-testid="vehicules-1"
    >
      VALUE_2
    </li>
  </ul>
</div>
`;
